controllers/statusreport/statusreport_adapter.go

```diff
@@ -86,3 +91,212 @@ func isFeatureEnabled() bool {
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	testStatuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(a.snapshot)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	allIntegrationTestsFinished, allIntegrationTestsPassed := a.determineIfAllIntegrationTestsFinishedAndPassed(integrationTestScenarios, testStatuses)
+	if err != nil {
+		a.logger.Error(err, "Failed to determine outcomes for Integration Tests",
+			"snapshot.Name", a.snapshot.Name)
+		return controller.RequeueWithError(err)
+	}
+
+	// Skip doing anything if not all Integration tests were finished for all integrationTestScenarios
+	if !allIntegrationTestsFinished {
+		a.logger.Info("Not all required Integration PipelineRuns finished",
+			"snapshot.Name", a.snapshot.Name)
+		return controller.ContinueProcessing()
+	}
+
+	// Since all integration tests finished, set the Snapshot Integration status as finished, but don't update the resource yet
+	gitops.SetSnapshotIntegrationStatusAsFinished(a.snapshot,
+		"Snapshot integration status condition is finished since all testing pipelines completed")
+	a.logger.LogAuditEvent("Snapshot integration status condition marked as finished, all testing pipelines completed",
+		a.snapshot, helpers.LogActionUpdate)
+
+	// If the Snapshot is a component type, check if the global component list changed in the meantime and
+	// create a composite snapshot if it did. Does not apply for PAC pull request events.
+	if metadata.HasLabelWithValue(a.snapshot, gitops.SnapshotTypeLabel, gitops.SnapshotComponentType) && !gitops.IsSnapshotCreatedByPACPullRequestEvent(a.snapshot) {
+		component, err := a.loader.GetComponentFromSnapshot(a.client, a.context, a.snapshot)
+		if err != nil {
+			a.logger.Error(err, "Failed to get Component for snapshot")
+			return controller.RequeueWithError(err)
+		}
+
+		compositeSnapshot, err := a.createCompositeSnapshotsIfConflictExists(a.application, component, a.snapshot)
+		if err != nil {
+			a.logger.Error(err, "Failed to determine if a composite snapshot needs to be created because of a conflict",
+				"snapshot.Name", a.snapshot.Name)
+			return controller.RequeueWithError(err)
+		}
+
+		if compositeSnapshot != nil {
+			a.logger.Info("The global component list has changed in the meantime, marking snapshot as Invalid",
+				"snapshot.Name", a.snapshot.Name)
+			if !gitops.IsSnapshotMarkedAsInvalid(a.snapshot) {
+				patch := client.MergeFrom(a.snapshot.DeepCopy())
+				gitops.SetSnapshotIntegrationStatusAsInvalid(a.snapshot,
+					"The global component list has changed in the meantime, superseding with a composite snapshot")
+				err := a.client.Status().Patch(a.context, a.snapshot, patch)
+				if err != nil {
+					a.logger.Error(err, "Failed to update the status to Invalid for the snapshot",
+						"snapshot.Name", a.snapshot.Name)
+					return controller.RequeueWithError(err)
+				}
```

Comment:
Can we have this as function call `MarkSnapshotAsInvalid`, to make it more readable and consistent with other statuses.